manifests/run.pp

```diff
@@ -120,39 +120,53 @@
   $service_name = $docker::service_name
   $docker_group = $docker::docker_group
 
-  validate_re($image, '^[\S]*$')
-  validate_re($title, '^[\S]*$')
-  validate_re($memory_limit, '^[\d]*(b|k|m|g)$')
-  validate_re($ensure, '^(present|absent)')
+  if $image {
+  assert_type(Pattern[/^[\S]*$/], $image)
```

Comment:
As above - I'd consider moving the params here to data type Pattern to be able to remove the assert_types here.